src/blas/backends/cublas/CMakeLists.txt

```diff
@@ -37,6 +37,10 @@ target_include_directories(${LIB_OBJ}
           ${PROJECT_SOURCE_DIR}/src
 )
 target_compile_options(${LIB_OBJ} PRIVATE ${ONEMKL_BUILD_COPT})
+target_compile_options(ONEMKL::SYCL::SYCL INTERFACE
```

Comment:
These options are what I was missing when I tried this before...

Reply:
@andrewtbarker  yes, because oneMKL cuBLAS backend does not have any kernel in it. No target triple was needed and it automatically using the DPCPP default target triple which is SPIR64 , This would generate the wrong binary code. once a kernel code is used inside the oneMKL backend we need to pass the relevant target triple according to the chosen device so that the correct binary would be generated